pkg/deployment/resources/pod_init_containers.go

```diff
@@ -0,0 +1,58 @@
+//
+// DISCLAIMER
+//
+// Copyright 2023 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+
+package resources
+
+import (
+	core "k8s.io/api/core/v1"
+
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1"
+	"github.com/arangodb/kube-arangodb/pkg/deployment/features"
+)
+
+// applyInitContainersResourceResources updates passed init containers to ensure that resources are set (if such feature is enabled)
+// This is applied only to containers added by operator itself
+func applyInitContainersResourceResources(initContainers []core.Container, mainContainerResources *core.ResourceRequirements) []core.Container {
+	if !features.InitContainerCopyResources().Enabled() || mainContainerResources == nil {
+		return initContainers
+	}
+
+	for i, c := range initContainers {
+		if !api.IsReservedServerGroupInitContainerName(c.Name) {
+			continue
+		}
+
+		if len(c.Resources.Limits) == 0 {
```

Comment:
Instead:

```
for k, v := range mainContainerResources {
  if _, ok := c.Resources.Limits[k]; !ok {
    c.Resources.Limits[k] = v
  
```

+ Dict init etc if needed

Reply:
Done. See linked PR for test results (still running)